lib/rspec/core/notifications.rb

```diff
@@ -496,7 +496,8 @@ def duplicate_rerun_locations
     # @attr duration [Float] the time taken (in seconds) to run the suite
     # @attr examples [Array<RSpec::Core::Example>] the examples run
     # @attr number_of_examples [Fixnum] the number of examples to profile
-    ProfileNotification = Struct.new(:duration, :examples, :number_of_examples)
+    # @attr profiler [RSpec::Core::Profiler] the profiler used to capture examples
+    ProfileNotification = Struct.new(:duration, :examples, :number_of_examples, :profiler)
```

Comment:
So what's weird to me about putting the profiler object on here is that registered formatter objects receive notification objects in the various event method calls.  Here you've making a notification object which holds a reference to a formatter.  So you'll wind up passing a notification containing a formatter to another formatter...which seems weird, circular and confusing.

Why would a formatter have to gain access to another formatter via a notification?

I think the notification just needs to have the profiling result data, and not the formatter object itself, right?  Given that...would it be a better separation of concerns if the notification got a reference to the profiling results but not the profiler itself?